integration/query_comparision_test.go

```diff
@@ -15,48 +15,235 @@
 package integration
 
 import (
-	"fmt"
+	"math"
 	"testing"
+	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
 	"go.mongodb.org/mongo-driver/mongo/options"
 
 	"github.com/FerretDB/FerretDB/integration/shareddata"
 )
 
-func TestQueryComparisionEq(t *testing.T) {
+func TestQueryComparisonEq(t *testing.T) {
 	t.Parallel()
 	providers := []shareddata.Provider{shareddata.Scalars, shareddata.Composites}
 	ctx, collection := setup(t, providers...)
 
-	var docs []bson.D
-	for _, provider := range providers {
-		docs = append(docs, provider.Docs()...)
-	}
+	for name, tc := range map[string]struct {
+		q           bson.D
+		expectedIDs []any
+	}{
+		// $eq
+		"EqDouble": {
+			q:           bson.D{{"value", bson.D{{"$eq", 42.13}}}},
+			expectedIDs: []any{"double"},
+		},
+		"EqDoubleNegativeInfinity": {
+			q:           bson.D{{"value", bson.D{{"$eq", math.Inf(-1)}}}},
+			expectedIDs: []any{"double-negative-infinity"},
+		},
+		"EqDoublePositiveInfinity": {
+			q:           bson.D{{"value", bson.D{{"$eq", math.Inf(+1)}}}},
+			expectedIDs: []any{"double-positive-infinity"},
+		},
+		"EqDoubleMax": {
+			q:           bson.D{{"value", bson.D{{"$eq", math.MaxFloat64}}}},
+			expectedIDs: []any{"double-max"},
+		},
+		"EqDoubleSmallest": {
+			q:           bson.D{{"value", bson.D{{"$eq", math.SmallestNonzeroFloat64}}}},
+			expectedIDs: []any{"double-smallest"},
+		},
+		"EqDoubleZero": {
+			q:           bson.D{{"value", bson.D{{"$eq", 0.0}}}},
+			expectedIDs: []any{"double-negative-zero", "double-zero", "int32-zero", "int64-zero"},
+		},
+		"EqDoubleNaN": {
+			q:           bson.D{{"value", bson.D{{"$eq", math.NaN()}}}},
+			expectedIDs: []any{"double-nan"},
+		},
+		"EqNegativeZero": {
+			q:           bson.D{{"value", bson.D{{"$eq", math.Copysign(0, -1)}}}},
+			expectedIDs: []any{"double-negative-zero", "double-zero", "int32-zero", "int64-zero"},
+		},
+
+		"EqString": {
```

Comment:
empty line - another block of what?  here and below.

Reply:
The separation of types, I think.

But then the one above should be `EqDoubleNegativeZero`, with "Double" in the middle

Reply:
Another block of data type. If it's redundunt, I will delete empty lines.

Reply:
I don't have a strong opinion there, but maybe try to add them as separate top-level tests? One reason we want to move away from the previous handlers tests is that one huge table is hard to maintain

Reply:
well then empty lines might be useful when we'll think of dividing tests into groups and subgroups.